prefect_azure/aci.py

```diff
@@ -0,0 +1,335 @@
+        default=ACI_DEFAULT_CPU,
+        description=(
+            "The number of virtual CPUs to assign to the task container. "
+            f"If not provided, a default value of {ACI_DEFAULT_CPU} will be used."
+        ),
+    )
+    gpu_count: Optional[float] = Field(
+        title="GPU Count",
+        default=None,
+        description=(
+            "The number of GPUs to assign to the task container. "
+            f"If not provided, no GPU will be used."
+        ),
+    )
+    gpu_sku: Optional[str] = Field(
+        title="GPU SKU",
+        default=None,
+        description=(
+            "The Azure GPU SKU to use. See the ACI documentation for a list of "
+            "GPU SKUs available in each Azure region."
+        ),
+    )
+    memory: float = Field(
+        default=ACI_DEFAULT_MEMORY,
+        description=(
+            "The amount of memory in gigabytes to provide to the ACI task. Valid amounts are "
+            "specified in the Azure documentation. If not provided, a default value of "
+            f"{ACI_DEFAULT_MEMORY} will be used unless present on the task definition."
+        ),
+    )
+    stream_output: bool = Field(
+        default=None,
+        description=(
+            "If `True`, logs will be streamed from the Prefect container to the local "
+            "console. Unless you have configured AWS CloudWatch logs manually on your "
+            "task definition, this requires the same prerequisites outlined in "
+            "`configure_cloudwatch_logs`."
+        ),
+    )
+    env: Dict[str, Optional[str]] = Field(
+        title="Environment Variables",
+        default_factory=dict,
+        description=(
+            "Environment variables to provide to the task run. These variables are set "
+            "on the Prefect container at task runtime. These will not be set on the "
+            "task definition."
+        ),
+    )
+    # Execution settings
+    task_start_timeout_seconds: int = Field(
+        default=240,
+        description=(
+            "The amount of time to watch for the start of the ACI task "
+            "before marking it as failed. The task must enter a RUNNING state to be "
+            "considered started."
+        ),
+    )
+    task_watch_poll_interval: float = Field(
+        default=5.0,
+        description=(
+            "The amount of time to wait between Azure API calls while monitoring the "
+            "state of an Azure Container Instances task."
+        ),
+    )
+
+    @sync_compatible
+    async def run(self, task_status: Optional[TaskStatus] = None) -> ACITaskResult:
+        """
+        Run the configured task on ACI.
+        """
+        if not self.command:
+            raise ValueError("Container cannot be run with empty command.")
+
+        # TODO: determine how to make DefaultAzureCredential work as expected
+        # if self.azure_credentials:
+        #     self.azure_credentials.login()
+        # token_credential = DefaultAzureCredential
```

Comment:
I went with service principal authentication for now, since that's what we're using with some other Azure blocks and it seems to meet users needs. I haven't seen any community requests to use managed identity instead of a service principal, so perhaps it's best to stick with what we know people are using?

Reply:
Yup, that sounds good